### PR TITLE
WIP: After finals dhaka

### PR DESCRIPTION
**Pull request by troy2914**
bits learned.

### CODE REVIEW

**Comment by vmcj** [comment]
I'll take a look next week, as this PR is large and I have a contest this weekend.

File: ansible.cfg [hunk 1]
```diff
@@ -1,8 +1,13 @@
 [defaults]
 inventory = ./hosts.yml
-forks = 50
+forks = 150
 host_key_checking = False
 pipelining = True
+log_path = /tmp/ansbile.log
+display_args_to_stdout = True
+# this makes debugging harder,
+# but allows hosts to advance indepently
```
Comment (vmcj) [inline ansible.cfg:9]:
```suggestion
# but allows hosts to advance independently
```

File: disable_challenge.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+---
+- name: configure contestant machines to undo challenge
```
Comment (vmcj) [inline disable_challenge.yml:2]:
```suggestion
- name: Configure contestant machines to undo challenge
```

Als I think I would prefer one challenge runbook which can be run with a tag `on`,`off` to switch between the 2 modes. I'll take a look if this is doable with the current books.

File: disable_challenge.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  hosts: contestants
+  gather_facts: false
+
+  tasks:
+    - name: restore firefox to default
```
Comment (vmcj) [inline disable_challenge.yml:7]:
```suggestion
    - name: Restore firefox to default
```

File: disable_challenge.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    - name: disable DNS bits in resolved.conf
+      ansible.builtin.replace:
+        path: /etc/systemd/resolved.conf
+        regexp: '^[#]?DNS=.*'
+        replace: '#DNS='
+        owner: root
+        group: root
+        mode: '0644'
+
+    - name: enable systemd-resolved
```
Comment (vmcj) [inline disable_challenge.yml:21]:
```suggestion
    - name: Disable systemd-resolved
```

File: disable_challenge.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        protocol: tcp
+        destination: 10.3.3.215/32
+        destination_ports:
+          - "80"
+          - "443"
+          - "53"
+        jump: drop-n-log
+        action: insert
+
+    - name: firewall blocks \#1
```
Comment (vmcj) [inline disable_challenge.yml:41]: I suspect we can do a loop here on `destination`

File: files/challenge.firefox.desktop [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+Keywords[pt_BR]=Internet;WWW;Browser;Web;Explorador;Navegador
+Keywords[ru]=Internet;WWW;Browser;Web;Explorer;интернет;браузер;веб;файрфокс;огнелис
+Keywords[sk]=Internet;WWW;Prehliadač;Web;Explorer
+Keywords[sl]=Internet;WWW;Browser;Web;Explorer;Brskalnik;Splet
+Keywords[tr]=İnternet;WWW;Tarayıcı;Web;Gezgin;Web sitesi;Site;sörf;çevrimiçi;tara
+Keywords[uk]=Internet;WWW;Browser;Web;Explorer;Інтернет;мережа;переглядач;оглядач;браузер;веб;файрфокс;вогнелис;перегляд
+Keywords[vi]=Internet;WWW;Browser;Web;Explorer;Trình duyệt;Trang web
+Keywords[zh_CN]=Internet;WWW;Browser;Web;Explorer;网页;浏览;上网;火狐;Firefox;ff;互联网;网站;
+Keywords[zh_TW]=Internet;WWW;Browser;Web;Explorer;網際網路;網路;瀏覽器;上網;網頁;火狐
+Exec=firefox https://icpcdhaka.contest.codeforces.com/
```
Comment (vmcj) [inline files/challenge.firefox.desktop:153]: If we template this one we can easier have different scenarios. I think the desktop files are the same apart from the URL here.

File: ip_link_show.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---
+- name: gather ip link addresseses
```
Comment (vmcj) [inline ip_link_show.yml:2]: I would suspect we get this data already if we would gather facts? Why can't we use that one?

File: lastminute.yml [hunk 8]
```diff
@@ -126,9 +160,90 @@
...
+      ansible.builtin.copy:
+        src: 'files/"{{ contest_detail }}".TechNotes."{{ papersize }}".pdf'
+        dest: /usr/share/doc/icpc/TechNotes.pdf
+      when: technotes.stat.exists
+
+    - name: copy domjudge TeamGuide.pdf (if ccs == domjudge)
+      ansible.builtin.copy:
+        src: files/"{{ contest_detail }}-domjudge-team-manual.pdf"
+        dest: /usr/share/doc/icpc/TeamGuide.pdf
+      when: '"{{ ccs }}" == "domjudge"'
```
Comment (vmcj) [inline lastminute.yml:188]: If you put these things in a block

``` 
- name: DOMjudge only bits
  block:
     < The tasks with the when: '"{{ ccs }}" == "domjudge"' >
     ``` we get less repitition

File: roles/coachview/vars/main.yml [hunk 9]
```diff
@@ -0,0 +1,4 @@
+---
+coachview_cds_url: https://172.29.1.207/api/contests/
```
Comment (vmcj) [inline roles/coachview/vars/main.yml:2]: I think we should put this file in the .gitignore as we already have a defaults which documents the variables.

**Review by vmcj — CHANGES_REQUESTED** [review]
I think reviewing would be easier if this PR is split over:
what to do for the windows boxes
what to do to setup & reset the challenge
the rest.

File: ip_link_show.yml (hunk 7 above)
Comment (troy2914) [reply ip_link_show.yml]: I believe this script's intent was to the get the mac address, i have updated the overall name field.

File: roles/coachview/vars/main.yml (hunk 9 above)
Comment (troy2914) [reply roles/coachview/vars/main.yml]: git rm'd it and the iptablesrules one.

Comment (troy2914) [reply roles/coachview/vars/main.yml]: can we put a roles/*/var/main.yml in the .gitignore?

**Pull request closed without merging**